io/aligned_buf_pool.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/cleanup/cleanup.h>
+
+#include <deque>
+#include <memory>
+#include <optional>
+
+#include "io/io.h"
+
+namespace io {
+
+// Buffer aligned to 4kb that is optionally backed by a registered buffer (from register_buffers)
+struct AlignedBuf {
+  AlignedBuf(MutableBytes buf, std::optional<unsigned> registered_idx = std::nullopt)
+      : buf(buf), registered_idx(registered_idx) {
+  }
+
+  const MutableBytes buf;                        // buf, nbytes
+  const std::optional<unsigned> registered_idx;  // optional buf_index for prep_read/write_fixed
+};
+
+// Hands out cunks of a large registered buffer, falls back to allocating non-registered
+// chunk if no free space is available.
+struct AlignedBufPool {
+  const static size_t kAlign = 4096;
+
+  // Request aligned buffer. Returns buffer and cleanup that recycles the buffer once dropped.
+  static auto Request(size_t length) {
+    AlignedBuf buf = RequestChunk(length);
+    return std::make_pair(buf, absl::MakeCleanup([ptr = buf.buf.data()] { ReturnChunk(ptr); }));
+  }
+
+  // Initialize buffer on current thread and register backing in io_uring
```

Comment:
fibers2 lib depends on io lib, not the other way around. it's better to move this file into fibers dir